patroni/api.py

```diff
@@ -202,13 +202,15 @@ def __init__(self, patroni, config):
     def query(self, sql, *params):
         cursor = None
         try:
-            with self.patroni.postgresql.connection().cursor() as cursor:
-                cursor.execute(sql, params)
-                return [r for r in cursor]
+            cursor = self.patroni.postgresql.connection().cursor()
```

Comment:
The with support for the cursor was introduced in psycopg2 2.5, which was released 2.5 years ago, so it's not an unreasonable requirement (good point is that we should actually formalise it in requirements).